tests/spawn-trace-viewer.spec.ts

```diff
@@ -0,0 +1,184 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { enableConfigs, expect, selectConfig, selectTestItem, test, traceViewerInfo } from './utils';
+
+test.beforeEach(({ showBrowser }) => {
+  test.skip(showBrowser);
+  // prevents spawn trace viewer process from opening in browser
+  process.env.PWTEST_UNDER_TEST = '1';
+});
+
+test.use({ showTrace: true, envRemoteName: 'ssh-remote' });
+
+test('@smoke should open trace viewer', async ({ activate }) => {
+  const { vscode, testController } = await activate({
+    'playwright.config.js': `module.exports = { testDir: 'tests' }`,
+    'tests/test.spec.ts': `
+      import { test } from '@playwright/test';
+      test('should pass', async () => {});
+    `,
+  });
+
+  await testController.run();
+  await testController.expandTestItems(/test.spec/);
+  selectTestItem(testController.findTestItems(/pass/)[0]);
+
+  await expect.poll(() => traceViewerInfo(vscode)).toMatchObject({
+    type: 'spawn',
+    traceFile: expect.stringContaining('pass'),
+  });
+});
+
+test('should change opened file in trace viewer', async ({ activate }) => {
+  const { vscode, testController } = await activate({
+    'playwright.config.js': `module.exports = { testDir: 'tests' }`,
+    'tests/test.spec.ts': `
+      import { test } from '@playwright/test';
+      test('one', async () => {});
+      test('two', async () => {});
+    `,
+  });
+
+  await testController.run();
+  await testController.expandTestItems(/test.spec/);
+
+  selectTestItem(testController.findTestItems(/one/)[0]);
+
+  await expect.poll(() => traceViewerInfo(vscode)).toMatchObject({
+    type: 'spawn',
+    traceFile: expect.stringContaining('one'),
+  });
+
+  selectTestItem(testController.findTestItems(/two/)[0]);
+
+  await expect.poll(() => traceViewerInfo(vscode)).toMatchObject({
+    type: 'spawn',
+    traceFile: expect.stringContaining('two'),
+  });
+});
+
+test('should not open trace viewer if test did not run', async ({ activate }) => {
+  const { vscode, testController } = await activate({
+    'playwright.config.js': `module.exports = { testDir: 'tests' }`,
+    'tests/test.spec.ts': `
+      import { test } from '@playwright/test';
+      test('should pass', async () => {});
+    `,
+  });
+
+  await testController.expandTestItems(/test.spec/);
+  selectTestItem(testController.findTestItems(/pass/)[0]);
+
+  await expect.poll(() => traceViewerInfo(vscode)).toBeUndefined();
+});
+
+test('should refresh trace viewer while test is running', async ({ activate }) => {
+  const { vscode, testController } = await activate({
+    'playwright.config.js': `module.exports = { testDir: 'tests' }`,
+    'tests/test.spec.ts': `
+      import { test } from '@playwright/test';
+      test('should pass', async () => await new Promise(r => setTimeout(r, 1000)));
```

Comment:
1000 won't be enough on CI. It is best to implement a latch here, we have it in ui mode test harness, so you can copy and reuse it here: https://github.com/microsoft/playwright/blob/main/tests/playwright-test/ui-mode-test-progress.spec.ts#L124